test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -8,6 +8,537 @@ test.describe( 'Links', () => {
+					content:
+						'This is Gutenberg: <a href="https://wordpress.org/gutenberg">https://wordpress.org/gutenberg</a>',
+				},
+			},
+		] );
+	} );
+
+	test( `can be created instantly when a URL is selected`, async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type(
+			'This is Gutenberg: https://wordpress.org/gutenberg'
+		);
+
+		// Select the URL.
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft', { times: 7 } );
+
+		// Click on the Link button.
+		await page.getByRole( 'button', { name: 'Link' } ).click();
+
+		// A link with the selected URL as its href should have been inserted.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content:
+						'This is Gutenberg: <a href="https://wordpress.org/gutenberg">https://wordpress.org/gutenberg</a>',
+				},
+			},
+		] );
+	} );
+
+	test( `is not created when we click away from the link input`, async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'This is Gutenberg' );
+
+		// Select some text.
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+
+		// Click on the Link button.
+		await page.getByRole( 'button', { name: 'Link' } ).click();
+
+		// Type a URL.
+		await page.keyboard.type( 'https://wordpress.org/gutenberg' );
+
+		// Click somewhere else - it doesn't really matter where.
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.focus();
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is Gutenberg',
+				},
+			},
+		] );
+	} );
+
+	test( `can be edited`, async ( { page, editor, pageUtils, LinkUtils } ) => {
+		await LinkUtils.createAndReselectLink();
+
+		// Click on the Edit button.
+		await page.getByRole( 'button', { name: 'Edit' } ).click();
+
+		// Change the URL.
+		await page.getByPlaceholder( 'Search or type url' ).fill( '' );
+		await page.keyboard.type( '/handbook' );
+
+		// Submit the link.
+		await pageUtils.pressKeys( 'Enter' );
+
+		// The link should have been updated.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is <a href="/handbook">Gutenberg</a>',
+				},
+			},
+		] );
+	} );
+
+	test( `can be removed`, async ( { page, editor, LinkUtils } ) => {
+		await LinkUtils.createAndReselectLink();
+
+		// Unlick via shortcut
+		// we do this to avoid an layout edge case whereby
+		// the rich link preview popover will obscure the block toolbar
+		// under very specific circumstances and screensizes.
+		await page
+			.locator( '.block-editor-link-control__search-item-top' )
+			.getByRole( 'button', { name: 'Unlink' } )
+			.click();
```

Comment:
...but this doesn't use a shortcut 😕

Reply:
I'm going to remove the comment on the basis that this scenario is probably not a problem for Playwright like it was when these tests were originally written for Puppeteer. 
```suggestion
		await page
			.locator( '.block-editor-link-control__search-item-top' )
			.getByRole( 'button', { name: 'Unlink' } )
			.click();
```